charts/simple-app/templates/deployment.yaml

```diff
@@ -79,6 +81,9 @@ spec:
   {{- end }}
   {{- if and $.Values.replicaCount (not $.Values.autoscaling.enabled) }}
   replicas: {{ $.Values.replicaCount }}
+  {{- else if $deactivateHpaForMaintenance }}
+  {{- /* https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#implicit-maintenance-mode-deactivation */}}
+  replicas: 0
```

Comment:
Wait ... I see the docs, but wouldn't this automatically scale the `Deployment` down? Have  you tested this behavior?

Reply:
Looking at metrics of the deployment when we ran tests, it _**did**_ scale the `Deployment`  down:

<img width="702" alt="Screenshot 2024-09-25 at 7 30 55 PM" src="https://github.com/user-attachments/assets/417e7de8-7a44-479a-b808-1f0f4897ff78">

As there are no documented ways to "pause" it (besides finagling with `behavior`), we _can_ try to use this approach mentioned in this comment: https://github.com/kubernetes/kubernetes/issues/28603#issuecomment-990507193

Its implementation is at https://github.com/Nextdoor/k8s-charts/pull/338/commits/d07c6c8902288ba5a67eb3abc9492346131f74fe

Reply:
> besides finagling with behavior

Tested that we do _**not**_ indeed have to `merge` the behavior keys that come in - squashing in maintenance mode is good enough: https://github.com/Nextdoor/k8s-charts/pull/338/commits/0302e28110f68048d171a5d6ac668984f033b539